### PR TITLE
Object.isEqual() & String.isEqual() dont require `type` argument now

### PR DESCRIPTION
**Pull request by amawdiin**
Object.isEqual() will now only be given 4 arguments, 'source1, path1, source2 path2' 
and 
String.isEqual() now only be given 3 arguments `source, path, string` 
This is done for better readability and ease of use. All tests run succesfully.

### CODE REVIEW

File: src/jmc/compile/command/builtin_function/bool_function.py [hunk 1]
```diff
@@ -25,6 +25,19 @@ def call_bool(self) -> tuple[str, bool, list[str]]:
         return f'score {self.args["selector"]} {self.args["objective"]} matches 1..', UNLESS, [
         ]
 
+# please put this somewhere in appropriate place, idk where to put these 2 functions so for now they're here.
+class nbtSource:
```
Comment (WingedSeal) [inline src/jmc/compile/command/builtin_function/bool_function.py:29]: Class with no instantiation

File: src/jmc/compile/command/builtin_function/bool_function.py [hunk 2]
```diff
@@ -25,6 +25,19 @@ def call_bool(self) -> tuple[str, bool, list[str]]:
         return f'score {self.args["selector"]} {self.args["objective"]} matches 1..', UNLESS, [
         ]
 
+# please put this somewhere in appropriate place, idk where to put these 2 functions so for now they're here.
+class nbtSource: 
+    def is_uuid(string) -> str:
```
Comment (WingedSeal) [inline src/jmc/compile/command/builtin_function/bool_function.py:30]: No type hint for string, wrong type hint for return value

File: src/jmc/compile/command/builtin_function/bool_function.py [hunk 3]
```diff
@@ -25,6 +25,19 @@ def call_bool(self) -> tuple[str, bool, list[str]]:
...
+        parts = string.split('-')
+        return len(parts) == 5 and all(len(part) in (8, 4, 4, 4, 12) and part.isalnum() for part in parts)
+    def get_source_type(source) -> str:
```
Comment (WingedSeal) [inline src/jmc/compile/command/builtin_function/bool_function.py:33]: No type hint
Lack of string documentation

File: src/jmc/compile/command/builtin_function/bool_function.py [hunk 4]
```diff
@@ -25,6 +25,19 @@ def call_bool(self) -> tuple[str, bool, list[str]]:
...
+        if source.startswith("@") or ObjectIsEqual.is_uuid(source):
```
Comment (WingedSeal) [inline src/jmc/compile/command/builtin_function/bool_function.py]: Wrong use of class method

File: src/jmc/compile/command/builtin_function/bool_function.py [hunk 5]
```diff
@@ -25,6 +25,19 @@ def call_bool(self) -> tuple[str, bool, list[str]]:
...
+            source_type = "entity"
+        elif source[0] in "~^" or source.isnumeric():
+            source_type = "block"
+        else:
+            source_type = "storage"
+        return source_type
```
Comment (WingedSeal) [inline src/jmc/compile/command/builtin_function/bool_function.py:40]: Unnecessary value for returning
Guard Clause is preferred

File: src/jmc/compile/command/builtin_function/bool_function.py [hunk 6]
```diff
@@ -43,16 +56,12 @@ class StringIsEqual(JMCFunction):
     def call_bool(self) -> tuple[str, bool, list[str]]:
         bool_result = self.datapack.data.get_current_bool_result()
         source = self.args["source"]
-        if self.args["type"] not in {"storage", "block", "entity"}:
-            raise JMCValueError(
-                f"'type' parameter expect 'storage' or 'block' or 'entity' (got {self.args['type']})",
-                self.raw_args["type"].token,
-                self.tokenizer)
-        if self.args["type"] == "storage" and ":" not in self.args["source"]:
+        type = nbtSource.get_source_type(source)
```
Comment (WingedSeal) [inline src/jmc/compile/command/builtin_function/bool_function.py:59]: Overriding built in variable (`type`)

**Review by WingedSeal — CHANGES_REQUESTED** [review]

File: src/jmc/compile/command/builtin_function/bool_function.py (hunk 4 above)
Comment (amawdiin) [reply src/jmc/compile/command/builtin_function/bool_function.py]: fixed

File: src/jmc/compile/command/builtin_function/bool_function.py (hunk 6 above)
Comment (amawdiin) [reply src/jmc/compile/command/builtin_function/bool_function.py]: fixed

File: src/jmc/compile/command/builtin_function/bool_function.py (hunk 5 above)
Comment (amawdiin) [reply src/jmc/compile/command/builtin_function/bool_function.py]: that has to be it, i think i fixed it.

File: src/jmc/compile/command/builtin_function/bool_function.py (hunk 1 above)
Comment (amawdiin) [reply src/jmc/compile/command/builtin_function/bool_function.py]: sorry i actually never used classes in python correctly, i had to do some research and now i guess it is what you wanted?

File: src/jmc/compile/command/builtin_function/bool_function.py [hunk 7]
```diff
@@ -25,12 +26,51 @@ def call_bool(self) -> tuple[str, bool, list[str]]:
         return f'score {self.args["selector"]} {self.args["objective"]} matches 1..', UNLESS, [
         ]
 
+class NbtSource:
```
Comment (WingedSeal) [inline src/jmc/compile/command/builtin_function/bool_function.py:29]: There's no use for it to be a class here.

File: src/jmc/compile/command/builtin_function/bool_function.py [hunk 8]
```diff
@@ -25,12 +26,51 @@ def call_bool(self) -> tuple[str, bool, list[str]]:
...
+    """
+    def __init__(self: str, source: str):
+        """Initializes a new instance of the NbtSource class.
+
+        Args:
+            source (str): The source of the data.
+        """
+        self.source = source
+
+    def is_uuid(source: str) -> bool:
```
Comment (WingedSeal) [inline src/jmc/compile/command/builtin_function/bool_function.py:41]: Make this a function

File: src/jmc/compile/command/builtin_function/bool_function.py [hunk 9]
```diff
@@ -25,12 +26,51 @@ def call_bool(self) -> tuple[str, bool, list[str]]:
...
+        """Checks if the given string is a UUID.
+
+        Args:
```
Comment (WingedSeal) [inline src/jmc/compile/command/builtin_function/bool_function.py:44]: Please do not use different docstring format

File: src/jmc/compile/command/builtin_function/bool_function.py [hunk 10]
```diff
@@ -25,12 +26,51 @@ def call_bool(self) -> tuple[str, bool, list[str]]:
...
+            source (str): The string to check.
+
+        Returns:
+            bool: True if the string is a UUID; otherwise, False.
+        """
+        parts = source.split('-')
+        return len(parts) == 5 and all(len(part) in (8, 4, 4, 4, 12) and part.isalnum() for part in parts)
+    
+    def get_type(self) -> str:
```
Comment (WingedSeal) [inline src/jmc/compile/command/builtin_function/bool_function.py:53]: Make this a function.

**Review by WingedSeal — CHANGES_REQUESTED** [review]
Almost good
Just a little adjustment. I was busy with final exam so I was very late to checking it.

**Pull request merged**